### ISSUE TITLE
Translate 2019 content to Portuguese

### ISSUE THREAD

**Issue by ibrahimcesar** [HTTPArchive/almanac.httparchive.org#505]
These are the core templates - without which we cannot release any translated chapters. They are in the [language specific templates directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/templates):
- [ ] base.html - 
- [ ] base_chapter.html- 
- [ ] contributors.html - 
- [ ] contributors.html (2020) - 
- [ ] error.html- 
- [ ] index.html - 
- [ ] index.html (2020) - 
- [ ] table_of_contents.html - 

These are the chapters to be translated, in rough order of popularity. They exist in the [content directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content):
- [ ] javascript - 
- [ ] css - 
- [ ] seo - 
- [ ] performance - 
- [ ] http2 - 
- [ ] markup - 
- [ ] third-parties -
- [ ] fonts - 
- [ ] accessibility -
- [ ] pwa - 
- [ ] cms - 
- [ ] security - 
- [ ] media - @garcaplay
- [ ] mobile-web - 
- [ ] cdn - 
- [x] ecommerce - @JMPerez 
- [ ] page-weight - 
- [ ] caching - 
- [ ] compression - 
- [ ] resource-hints - 

Additionally the following pages need translated too in the [language specific templates directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/templates):
- [ ] accessibility_statement.html - 
- [ ] base_ebook.html- 
- [ ] methodology.html - 

There is no need to translate the chapters HTML pages are they are generated off the markdown combined with the above templates.

Please include "Makes progress on #505 " in all pull requests so a link is created from the PR to this issue.

---

Common notes for writing consistency are here: https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Translators'-Guide. Feel free to edit that and/or add Portuguese-specific extras by editing this comment.

Portuguese-specific extra advice:
- None yet

**Comment by bazzadp** [comment]
There's been no progress on this since the issue was opened so am going to close this. We can reopen if we get translators.

@ibrahimcesar let us know if you plan on actioning this and want to re-open it?

**Comment by HakaCode** [comment]
would love to start at this point with `base.html`

**Comment by bazzadp** [comment]
> would love to start at this point with `base.html`

Go for it! Give me a shout if any questions.

What we’ve found best is to put your name beside the file you’re working on in the first comment of this issue. I’ve done that for base.html for now. I’ve sent you an invite to join the Translation team in GitHub so you will be able to edit the comment yourself once you accept that.

**Comment by HakaCode** [comment]
@bazzadp how to proceed with the next files to be translated? do I ask for permission here or can I just put my name next to the file I'm working on?

**Comment by bazzadp** [comment]
> @bazzadp how to proceed with the next files to be translated? do I ask for permission here or can I just put my name next to the file I'm working on?

No need to ask for permission - just stick your name against an unassigned file and off you go.

In fact, as you are the only Portuguese translator at present there's definitely no need to ask for permission. But let's follow the assigning workflow, so if anyone else joins you, then they can do the same!

**Comment by HakaCode** [comment]
> > @bazzadp como proceder com os próximos arquivos a serem traduzidos? peço permissão aqui ou posso apenas colocar meu nome ao lado do arquivo no qual estou trabalhando?
> 
> Não há necessidade de pedir permissão - basta colar seu nome em um arquivo não atribuído e pronto.
> 
> Na verdade, como você é o único tradutor de português no momento, definitivamente não há necessidade de pedir permissão. Mas vamos seguir o fluxo de trabalho de atribuição, então, se outra pessoa se juntar a você, eles podem fazer o mesmo!

Thank you for your attention, I'm really enjoying the project.

**Comment by HakaCode** [comment]
@elaynelemos.
Caso se interesse, poderia me ajudar a traduzir? 👍

**Comment by elaynelemos** [comment]
> @elaynelemos.
> Caso se interesse, poderia me ajudar a traduzir?

Opa! Posso tentar cotribuir. Peguei algumas coisas para o hacktoberfest, mas posso incluir isso na minha TO DO list. haha

**Comment by HakaCode** [comment]
> > @elaynelemos.
> > Caso se interesse, poderia me ajudar a traduzir?
> 
> Opa! Posso tentar cotribuir. Peguei algumas coisas para o hacktoberfest, mas posso incluir isso na minha TO DO list. haha

Ficarei Grato!
Verdade ta rolando o Hacktoberfest nem lembrei disso, tbm preciso pegar umas coisinhas 👍.

**Comment by bazzadp** [comment]
@HakaCode just in case you missed this message from @rviscomi on Slack: https://httparchive.slack.com/archives/CGFLV2S06/p1601221746011800?thread_ts=1601197715.010600&cid=CGFLV2S06

> @HakaCode congrats on the launch and thank you for your translation efforts! i'd love to tweet about this from the HTTP Archive account. could you take a look at this Google Translate and let me know if there's a better way to say it?
>> Announcing the sixth and newest language of the Web Almanac, Portuguese!
>> Anunciamos o sexto e mais novo idioma do Web Almanac, o português!

**Comment by HakaCode** [comment]
> @HakaCode just in case you missed this message from @rviscomi on Slack: https://httparchive.slack.com/archives/CGFLV2S06/p1601221746011800?thread_ts=1601197715.010600&cid=CGFLV2S06
> 
> > @HakaCode congrats on the launch and thank you for your translation efforts! i'd love to tweet about this from the HTTP Archive account. could you take a look at this Google Translate and let me know if there's a better way to say it?
> > > Announcing the sixth and newest language of the Web Almanac, Portuguese!
> > > Anunciamos o sexto e mais novo idioma do Web Almanac, o português!

Oops! excuse @bazzadp.
I forgot to join the slack, I barely know how to interact with the platform yet.
About the translation, this perfect!

**Comment by elaynelemos** [comment]
@HakaCode, posso pegar qualquer doc que não esteja marcado e sem PR?

**Comment by HakaCode** [comment]
> @HakaCode, posso pegar qualquer doc que não esteja marcado e sem PR?

@elaynelemos exato, antes coloque seu nome na frente do arquivo q for traduzir conforme os meus.

**Comment by elaynelemos** [comment]
> > @HakaCode, posso pegar qualquer doc que não esteja marcado e sem PR?
> 
> @elaynelemos exato, antes coloque seu nome na frente do arquivo q for traduzir conforme os meus.

Não tenho como alterar na issue. hahaha
Podes me marcar no http2. É a que estou trabalhando agora.

**Comment by HakaCode** [comment]
> Não tenho como alterar na issue. hahaha
> Podes me marcar no http2? É a que estou trabalhando agora.

@elaynelemos desculpe a demora, ja anotei, qualquer duvida so me chamar!

@bazzadp, @rviscomi.
 how to add @elaynelemos  as a translator?

**Comment by rviscomi** [comment]
Just created a new @HTTPArchive/portuguese-translators subteam and sent @elaynelemos an invitation 👍

**Comment by elaynelemos** [comment]
> Just created a new @HTTPArchive/portuguese-translators subteam and sent @elaynelemos an invitation

Thank you guys!

**Comment by bazzadp** [comment]
@eduqg you should have an invitation in your inbox to join the Portuguese team. Once you accept that that you'll be able to edit [the first comment](https://github.com/HTTPArchive/almanac.httparchive.org/issues/505#issue-523991913) to put your name beside another chapter if you want.

**Comment by elaynelemos** [comment]
Hi, y'all! I had a lot of work at the end of the last year but now I'm back. Happy new year :D

Let's rock it @HakaCode !

**Comment by brunopulis** [comment]
@bazzadp thanks for include me in the team. I found some 2019 translations that can be improved, how do I go about correcting them?

**Comment by bazzadp** [comment]
> @bazzadp thanks for include me in the team. I found some 2019 translations that can be improved, how do I go about correcting them?

Just open a PR for any errors you see. Hopefully @HakaCode or @elaynelemos can review any changes.

**Comment by emanuelgsouza** [comment]
Hey everyone. I'm glad to participate this work. I would like to start the translations of 2020 year (starting with CSS page). So, I have a question: should I create a folder called `2020` in the `src/content/pt`? And, in this folder, create the corresponding `md` file?

Thanks

**Comment by bazzadp** [comment]
Yup that's it.

Also with you first translation PR you should add yourself to the `config/2020.json` file (alphabetical by first name), in the `translators` team.

**Comment by soulcorrosion** [comment]
Hi everyone, I'm gonna select a topic and start working on it if you don't mind.

**Comment by bazzadp** [comment]
Btw if anyone would like to translate the foreword that would be greatly appreciated! Unlike the chapters, it’s reasonably short so should be quicker to do.

It’s in [src/templates/en/2020/base.html](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/templates/en/2020/base.html) but is missing from [src/templates/pt/2020/base.html](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/templates/pt/2020/base.html) as wasn’t written when this was originally translated.

**Comment by emanuelgsouza** [comment]
> Btw if anyone would like to translate the 2020 foreword that would be greatly appreciated! Unlike the chapters, it’s reasonably short so should be quicker to do.
> 
> It’s in [src/templates/en/2020/base.html](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/templates/en/2020/base.html) but is missing from [src/templates/pt/2020/base.html](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/templates/pt/2020/base.html) as wasn’t written when this was originally translated.

I can get this

**Comment by ademarCardoso** [comment]
hi guys, i would like to help translate, how can i help?

**Comment by bazzadp** [comment]
Hi @ademarCardoso , accept that invite I just sent you. Then edit the first issue to put your name beside one of the chapters you’re interested in translating. Then fork the repo and copy the markdown file from `src/content/en/2020/` to `src/content/pt/2020` and translate and submit a pull request to get it accepted.

Any questions at all just shout.

And if you spot any other issues in either English or what we’ve already translated to Portuguese, then feel free to raise an issue or PR to fix those too!

**Comment by ademarCardoso** [comment]
Thank you man! I will try to contribute a lot to the project

**Comment by bazzadp** [comment]
What Portuguese should we use?

This has come up in #2030 as think so far we’ve used Brazilian Portuguese but now have our first Portuguese-Portuguese translation (I think - correct me if I’m wrong).

I wrote this in [the translators guide](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Translators'-Guide):

> If it helps I would advise the primary or most common dialect is used (e.g. French spoken in France, rather than in Canada) - even if this means swallowing pride about the incorrect use of "American English" when they didn't even invent the language! 😁

Portuguese is similar to English in that the biggest speaking dialect is not the one the name suggests I presume. I guess the “primary” dialect is that spoken in Portugal but then is Brazilian more common because it’s so much larger? Or is that offset as other regions speak Portuguese Portuguese?

It also depends on the translator team make up - where are you all from?

 I note Wikipedia uses pt-br mostly, but not sure on the reason (I suspect volunteers so far speak Portuguese Brazilian). Also, on that subject, should we move our language code from pt to pt-br if we do decide that? We did this once before with Traditional Chinese so have ability to redirect and the like.

Anyway I really am guessing here so will hand over to you all to discuss and hopefully agree what to do here!

**Comment by emanuelgsouza** [comment]
@bazzadp , do we have metrics that where the users from? And, if the most of the translators are from Brazil, I believe we should use the Brazilian translation for now.

**Comment by bazzadp** [comment]
Here’s what GA is telling us:

![GA users with pt in language with pt-be being 81.93%](https://user-images.githubusercontent.com/10931297/109262238-1a7e9f00-77f9-11eb-8997-97a6e868dcce.jpeg)

**Comment by soulcorrosion** [comment]
Hey, I'm completely ok in staying pt-br. I think I'm the only portuguese in the team so, from my point of view, it wouldn't make much sense to go back and change all the chapters that were translated, so we need to keep going. It's just that any of our brazilians translators would have a bit more work aligning certain words from my PRs, but it would bem minor. The other way around would be impossible for me. :)

**Comment by bazzadp** [comment]
Good stuff. Decision made then. Updated first comment with this at the bottom:

> Portuguese-specific extra advice:
> - We use Brazilian Portuguese as more translators and visitors prefer that.

**Comment by rviscomi** [comment]
Is it worth explicitly noting the locale in the language code visible in the URL, similar to zh-TW? Or do we only do that for languages with multiple locales? For readers, it might be nice to set expectations, and for translators it may help reinforce the intended spelling (in addition to comments about best practices).

**Comment by bazzadp** [comment]
Yeah [I did suggest that above](https://github.com/HTTPArchive/almanac.httparchive.org/issues/505#issuecomment-786308615):

> I note Wikipedia uses pt-br mostly, but not sure on the reason (I suspect volunteers so far speak Portuguese Brazilian). Also, on that subject, should we move our language code from pt to pt-br if we do decide that? We did this once before with Traditional Chinese so have ability to redirect and the like.

But then never followed up on it!

I’m in two minds. Generally only do it if we have more dialects, but Brazilian Portuguese is sufficiently different that maybe should do it here?

As to precedence, Wikipedia doesn’t differentiate (but are they using Portuguese Portuguese?), but MDN does. But MDN also uses `en-US` and don’t think they support any other dialect of English.

So arguments for both sides, in which case maybe not worth the (admittedly small) hassle to move?

**Comment by HakaCode** [comment]
@elaynelemos 
I would like to know and check if the `2019 security` chapter has been translated, please?

**Comment by tunetheweb** [comment]
It might be better to look at the 2022 Security chapter instead anyway. I'm sure readers will be more intested in the latest report over the 2019 one.

**Comment by Fransuelton** [comment]
Hey guys, how can I help translate?

**Comment by tunetheweb** [comment]
Sure! Sent you an invite to the team so you can edit issues and tag yourself next to a chapter in [the first comment](https://github.com/HTTPArchive/almanac.httparchive.org/issues/505#issue-523991913). More details in the [Translator Guide](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Translators'-Guide).

Let me know if you have any questions!

**Comment by HakaCode** [comment]
> It might be better to look at the 2022 Security chapter instead anyway. I'm sure readers will be more intested in the latest report over the 2019 one.

@tunetheweb Good since someday we will get to translate the `2022 -security`, what would be your suggestion? focus on translating 2022 as quickly as possible or do we continue with 2019 and so on?

**Comment by tunetheweb** [comment]
> what would be your suggestion? focus on translating 2022 as quickly as possible

Yes. As I say I think readers are far more likely to want to read the latest than the previous years.

**Comment by mao1910** [comment]
"We use Brazilian Portuguese as more translators and visitors prefer that." given the sheer size of Brazil, that's understandable. I assume then that Portuguese PT translations are not to be used in order to avoid mixing the two language variants?

**Comment by tunetheweb** [comment]
Well we don’t like to turn down any contributions! But, given the only partial translations we’ve had so far, I’m not sure we have the the contributors to maintain two different sets of Portuguese. And mixing the two up would probably be more confusing.

But I’m not a Portuguese speaker, and also don’t know how different they are, so open to your thoughts here.
